src/scan-imports.ts

```diff
@@ -218,38 +218,55 @@ export async function scanImports(cwd: string, config: SnowpackConfig): Promise<
   }
 
   // Scan every matched JS file for web dependency imports
-  const loadedFiles = await Promise.all(
+  const loadedFiles: (SnowpackFile | null)[] = await Promise.all(
     includeFiles.map(async (filePath) => {
-      const ext = nodePath.extname(filePath);
+      const {baseExt, expandedExt} = getExt(filePath);
       // Always ignore dotfiles
       if (filePath.startsWith('.')) {
         return null;
       }
-      // Probably a license, a README, etc
-      if (ext === '') {
-        return null;
-      }
-      // Our import scanner can handle normal JS & even TypeScript without a problem.
-      if (ext === '.js' || ext === '.mjs' || ext === '.ts' || ext === '.jsx' || ext === '.tsx') {
-        return [filePath, await fs.promises.readFile(filePath, 'utf-8')] as [string, string];
-      }
-      if (ext === '.vue' || ext === '.svelte' || ext === '.html') {
-        const result = await fs.promises.readFile(filePath, 'utf-8');
-        // TODO: Replace with matchAll once Node v10 is out of TLS.
-        // const allMatches = [...result.matchAll(new RegExp(HTML_JS_REGEX))];
-        const allMatches: string[] = [];
-        let match;
-        const regex = new RegExp(HTML_JS_REGEX);
-        while ((match = regex.exec(result))) {
-          allMatches.push(match);
+
+      switch (baseExt) {
```

Comment:
This is an opinionated change. To me it just felt like we were dealing with enough extensions where a `switch()` felt easier to manage.